extendedcopy.go

```diff
@@ -170,3 +173,48 @@ func findRoots(ctx context.Context, storage content.GraphStorage, node ocispec.D
 	}
 	return roots, nil
 }
+
+// FilterArtifactType will configure opts.FindPredecessors to filter the predecessors
+// whose artifact type matches a given regex pattern. The regex syntax supported
+// is RE2. Reference: https://github.com/google/re2/wiki/Syntax.
+func (opts *ExtendedCopyGraphOptions) FilterArtifactType(pattern string) {
```

Comment:
Compiling regular expressions takes time. Therefore, it is better to reuse the compiled one by passing `filter *regexp.Regexp` instead of `pattern string`.
```suggestion
func (opts *ExtendedCopyGraphOptions) FilterArtifactType(filter *regexp.Regexp) {
```

Reply:
changed accordingly.